utilities/write_batch_with_index/write_batch_with_index.cc

```diff
@@ -450,6 +452,7 @@ bool WriteBatchWithIndex::Rep::UpdateExistingEntryWithCfId(
   }
   WriteBatchIndexEntry* non_const_entry =
       const_cast<WriteBatchIndexEntry*>(iter.GetRawEntry());
+  obsolete_offsets.push_back(non_const_entry->offset);
```

Comment:
Not familiar with WriteBatchWithIndex. When `UpdateExistingEntryWithCfId` will be called?

Reply:
After adding each entry to the batch. The name is kind of misleading. If there is no existing key it returns false, and then the callee will add the key.